sdk/test-utils/recorder-new/src/core-v2-recorder.ts

```diff
@@ -31,16 +33,25 @@ const paths = {
 export class TestProxyHttpClient {
   private url = "http://localhost:5000";
   public recordingId?: string;
-  public mode: string;
-  public httpClient: HttpClient;
+  public mode!: string;
+  public httpClient!: HttpClient;
   private stateManager = new RecordingStateManager();
-  private playback: boolean;
+  private playback!: boolean;
+  private sessionFile!: string;
 
-  constructor(private sessionFile: string) {
-    this.sessionFile = sessionFile;
-    this.mode = env.TEST_MODE;
-    this.playback = isPlaybackMode();
-    this.httpClient = createDefaultHttpClient();
+  constructor(private testContext?: Test | undefined) {
+    if (isRecordMode() || isPlaybackMode()) {
+      if (!this.testContext) {
```

Comment:
This is the nitpick of all nitpicks, but I think this will read slightly better if the success case is first and the error is in the `else`.